sylvia-derive/src/message.rs

```diff
@@ -921,9 +922,19 @@ impl<'a> GlueMessage<'a> {
         let interfaces_cnt = interface_names.len();
 
         let dispatch_arms = interfaces.iter().map(|interface| {
-            let ContractMessageAttr { variant, .. } = interface;
+            let ContractMessageAttr {
+                variant,
+                is_custom_msg,
+                ..
+            } = interface;
+
+            if *is_custom_msg && msg_ty == &MsgType::Exec
+            {
+                quote! { #contract_name :: #variant(msg) => Ok( #sylvia ::into_response::IntoResponse::into_response(msg.dispatch(contract, Into::into(ctx))?)?) }
+            } else {
+                quote! { #contract_name :: #variant(msg) => msg.dispatch(contract, Into::into(ctx)) }
+            }
```

Comment:
I don't like this decision. You introduce branching in every `interface`, while clearly, all interfaces are going to call the same branch.

It is not even a performance issue (technically might affect, but not really), but a readability issue - I need to think about it to understand that it is always the same branch taken.

Both of those ifs "bodies" shall be extracted to some function (even local closures), and then one should be selected before the iteration. The best is to maintain this function signature as `fn(&ContractMessageAttr) -> TokenStream` (not sure about result type - whatever is returned by `quote!`), then you simplify this to something like:

```rust
let emit_dispatch_arm = match *is_custom_msg && msg_ty == &MsgType::Exec {
  true => emit_dispatch_arm_resp_conversion,
  false => emit_default_dispatch_arm,
};
let dispatch_arms = interfaces.iter().map(emit_dispatch_arm);
```

Possibly even inlining the dispatch arms there.

Reply:
As discussed offline this is this is dispatch per interface and it's correct behavior. I can agree on changing it to match